src/NServiceBus.Core/ExceptionExtensions.cs

```diff
@@ -15,5 +16,9 @@ public static string GetMessage(this Exception exception)
                 return $"Could not read Message from exception type '{exception.GetType()}'.";
             }
         }
+
+#pragma warning disable PS0003 // A parameter of type CancellationToken on a non-private delegate or method should be optional
+        public static bool IsCausedBy(this Exception ex, CancellationToken cancellationToken) => ex is OperationCanceledException && cancellationToken.IsCancellationRequested;
```

Comment:
What do you think about this approach? It makes the purpose of the `when` filters painfully obvious in both common variants:

```c#
// in a regular method which just needs to do something when an operation fails for any reason
try
{
}
catch (Exception ex) when (!(ex is OperationCanceledException && cancellationToken.IsCancellationRequested))
{
}

// in an `...AndSwallowExceptions...` background task method
try
{
}
catch (OperationCanceledException ex) when (cancellationToken.IsCancellationRequested))
{
}
catch (Exception ex)
{
}
```

vs

```c#
// in a regular method which just needs to do something when an operation fails for any reason
try
{
}
catch (Exception ex) when (!ex.IsCausedBy(cancellationToken))
{
}

// in an `...AndSwallowExceptions...` background task method
try
{
}
catch (Exception ex) when (ex.IsCausedBy(cancellationToken)))
{
}
catch (Exception ex)
{
}
```

The end result is that this method is the _only_ place in the entire solution where `OperationCanceledException` is referenced.

If we go with this approach, then PS0019 can become just "don't catch OCE" and PS0020 won't be required.

Of course, each downstream will need its own `IsCausedBy` one-liner, if we stick to our policy of core not shipping general extensions/utilities.